enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -78,6 +78,23 @@ var dieOnFirecrackerFailure = flag.Bool("executor.die_on_firecracker_failure", f
 var workspaceDiskSlackSpaceMB = flag.Int64("executor.firecracker_workspace_disk_slack_space_mb", 2_000, "Extra space to allocate to firecracker workspace disks, in megabytes. ** Experimental **")
 
 const (
+	// goinitVersion determines the version of the go init binary that this
+	// executor supports. This version needs to be bumped when making
+	// incompatible changes to the goinit binary. This includes but is not
+	// limited to:
+	//
+	// - Adding new platform prop based features that depend on goinit support,
+	//   such as dockerd init options.
+	// - Adding new features to the vmexec.Exec service.
+	//
+	// We manually maintain this version instead of using a hash of the goinit
+	// binary because we expect the hash to be unstable, likely changing with
+	// each executor release.
+	//
+	// NOTE: this is part of the snapshot cache key, so bumping this version
+	// will make existing cached snapshots unusable.
+	goinitVersion = "0"
```

Comment:
What if we just use the hash of `//enterprise/vmsupport/bin/initrd.cpio`? It looks like that hash changes across changes to goinit, and then we can automatically invalidate snapshots if goinit changes. Let me know if that's going to be too sensitive to changes, or if I'm missing something, but would definitely prefer an automatic solution to this problem if we can find one.

Reply:
+1 to using something more automatic

I'm assuming the source file hash will cover the vast majority of cases. What if we just have people bump the remote instance name, the way we invalidate snapshots today, if they need to force an upgrade? 

This seems similar to the approach of manual version string + hash, but a little simpler to not maintain two arbitrary fields

Reply:
I suspect that the hash of `goinit` changes on every release, but let me check this by checking out some historical commits and seeing how often it changes.

Reply:
Sorry I forgot to respond to this earlier. I was talking about the "Run from clean workflow button" which updates the instance name on the snapshot key and invalidates former snapshots. I didn't think about the case where an executor could be incompatible with a version though, and I guess during deploys there could be a window where there are two incompatible executor versions running and we have to make sure they don't try to use a snapshot created by the other. So maybe we use a combination of the manual version string + a hash of some source files like you mentioned above?

Reply:
Thinking about it a bit more, I am a bit hesitant to include the goinit / vmexec source file hashes, because it would discourage us from making cleanups / refactorings that are backwards-compatible.

Let's start with a manually maintained version for now and if it bites us, we can switch to something better.

Reply:
Personally I still prefer the automatic approach 😅 I think false negatives where we forget to update the snapshot are going to be much more confusing to debug, and hopefully those source files won't change a lot when we've finalized this work. I could see the test becoming something that people ignore and end up bumping the version manually anyway. But I respect your decision to disagree haha

Reply:
I went with the test approach. This way we have options - we can lazily update the hash and bump the version if we don't want to think about it, but if we find that we're invalidating snapshots too often with this approach then we can be more careful about marking changes compatible vs incompatible.